src/amo/components/Addon/index.js

```diff
@@ -173,69 +193,127 @@ export class AddonBase extends React.Component {
         header={i18n.gettext('Rate your experience')}
         className="Addon-overall-rating"
         {...props}>
-        <RatingManager
-          addon={addon}
-          location={location}
-          version={addon.current_version}
-        />
+        {addon ?
+          <RatingManager
```

Comment:
I commented below–it looks like it *should* be straightforward.

But if it becomes a beast I'm happy to wait for another patch.